Implement a Script Pauser entity script (#307)
highfidelity/hifi-content: usefulUtilities/scriptPauser/README.md

# Script Pauser
When this script is attached to an entity as an entity script, the entity serves as a "Script Pauser", useful for when you don't want some script to be running in your domain.

## Features
- Script Pauser will stop a client script that a user is running if it meets one of the following criteria:
    1. The user-defined "fuzzy match" array contains a string that is within any script name that is currently running
        - CAUTION: Fuzzy matching script names comes with risks of accidentally stopping the wrong client script, resulting in more user frustration!
    2. The user-defined "exact match" array contains a string that exactly matches the script name that is currently running
- Any scripts that Script Pauser stops will be automatically restarted when Script Pauser is unloaded (i.e. the user leaves the domain in which the entity to which Script Pauser is attached is contained)

> NOTE that, when Script Pauser restores a Paused script that is a certified script, the "Installed" state of that script will be lost. This means that, when Script Pauser restores a previously-installed certified app, the Inventory app will not show that app as "Installed", even though its associated script is running.

Reviewer: NOTE: When Script Pauser restores a Paused script that is a certified script, the "Installed" state of that script will be lost. Meaning when Script Pauser restores a previously-installed certified app, the Inventory app will not show that app as "Installed", even though its associated script is running.

Author: i can't tell what you're changing here - could you please add this comment as a suggestion so that i can see the diff?

> NOTE: when Script Pauser restores a Paused script that is a certified script, the "Installed" state of that script will be lost; When Script Pauser restores a previously-installed certified app, the Inventory app will not show that app as "Installed", even though its associated script is running.

Reviewer: ```suggestion
NOTE: The Inventory App will lose the "Installed" state of a certified app after the Script Pauser pauses and restores the certified app. The certified app will be running, simply will not be seen as "Installed".
```
